### PR TITLE
[APM-3694] Add QPay Payment Request and Response source

### PR DESCRIPTION
**Pull request by andrew-berestov-cko**
# Description of changes in PR

Add QPay Payment Request and Response source

## Checklist

- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [x] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

**Review by alex-larin-cko — APPROVED** [review]

File: nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+description: 'QPay Source'
+allOf:
+  -
+    $ref: '#/components/schemas/PaymentResponseSource'
+  -
+    type: object
+    required:
+      - description
+      - pun
+    properties:
```
Comment (Enrica-Mameli-cko) [inline nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml:14]: in MBC we are not returning any of these data to the merchant
why do we have to do it now?
what the merchant will do with this information?
why should be passed in the source?

**Review by Enrica-Mameli-cko — CHANGES_REQUESTED** [review]

File: nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml (hunk 1 above)
Comment (Enrica-Mameli-cko) [reply nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml:14]: I am discussing with Anastasiya and we can probably generalise this

can you clarify what is the confirmation_id?
is this kind of an authentication or authorization id?

File: nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      description:
+        type: string
+        description: 'Alphanumeric string containing a description of the payment order.'
+      qpay_status:
```
Comment (Enrica-Mameli-cko) [inline nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml:18]: as agreed in today meeting `qpay_status ` should be returned in `processing` as `partner_status_code`

File: nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        type: string
+        description: 'The status code returned from the QPay gateway on payment, if available.'
+      status_message:
```
Comment (Enrica-Mameli-cko) [inline nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml:21]: as agreed in today meeting `status_message ` should be returned in `processing` as `partner_status_description`

File: nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        type: string
+        description: 'A message giving further detail on the payment status, for failure/cancelled/success status payments.'
+      confirmation_id:
```
Comment (Enrica-Mameli-cko) [inline nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml:24]: as agreed in today meeting `confirmation_id` should be returned in `processing` as `partner_ confirmation_id`

File: nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        type: string
+        description: 'An identifier from the QPay gateway for a successful payment.'
+      pun:
```
Comment (Enrica-Mameli-cko) [inline nas_spec/components/schemas/Payments/ResponseSources/PaymentResponseQPaySource.yaml:27]: as agreed in today meeting `pun` should be returned in `processing` as `partner_ payment_id`

**Review by Enrica-Mameli-cko — CHANGES_REQUESTED** [review]

**Review by Enrica-Mameli-cko — COMMENTED** [review]
please do not action the request changes until is clear if we are talking about the payment response or the get payment details response!

**Comment by tania-asimionesei-cko** [comment]
Please note that on the 27th of July we had a BREAKING CHANGE release, where the ABC and NAS versions of the API reference were switched in line with company-wide NAS readiness OKR and initiative. Therefore please make sure that you address any conflicts and updates in line with the latest release. 
PR for the NAS switch release can be found here: https://github.com/checkout/checkout-api-reference/pull/653

**Review by Enrica-Mameli-cko — COMMENTED** [review]
https://github.com/checkout/checkout-api-reference/pull/625#pullrequestreview-1043695655

according to the [payload agreement](https://docs.google.com/document/d/1ia5zhpJQKSpV403txMXlyeQXnw50CUajZ59BEoraUqs/edit) all those field should be in the GET response not in the payment response and I agree we cannot move anything in the payment response as it will break merchant integration. this is an old implementation and we will do an exception

**Comment by andre-lopes-cko** [comment]
Handled in this PR
https://github.com/checkout/checkout-api-reference/pull/671

**Pull request closed without merging**